src/vs/editor/browser/widget/multiDiffEditorWidget/diffEditorItemTemplate.ts

```diff
@@ -132,7 +133,15 @@ export class DiffEditorItemTemplate extends Disposable implements IPooledObject<
 		this._outerEditorHeight = 38;
 
 		this._register(this._instantiationService.createInstance(MenuWorkbenchToolBar, this._elements.actions, MenuId.MultiDiffEditorFileToolbar, {
-			actionRunner: this._register(new ActionRunnerWithContext(() => (this._viewModel.get()?.diffEditorViewModel?.model.modified.uri))),
+			actionRunner: this._register(new ActionRunnerWithContext(() => {
+				const modifiedUri = this._viewModel.get()?.diffEditorViewModel?.model.modified.uri;
+
+				if (typeof this._editorViewModel.get()?.getActionContext === 'undefined') {
+					return modifiedUri;
+				}
+
+				return modifiedUri ? this._editorViewModel.get()?.getActionContext!(modifiedUri) : undefined;
```

Comment:
I think the action args should be covariant* for every multi file diff editor instance. Otherwise it is impossible to implement commands that work on all instances.

*: Specialized multi file diff editor instances should use specialized action args, specialized meaning that a general handler doesn't notice that it is specialized.